### PR TITLE
Guus/pick multi

### PR DESCRIPTION
**Pull request by guusw**

### CODE REVIEW

File: include/shards/shards.h [hunk 1]
```diff
@@ -590,6 +590,8 @@ struct SHVarPayload {
 // used internally to cancel and abort a wire flow, must be String type and receiver must abort and destroy the var
 // used only in async.hpp and rust side async activation so far
 #define SHVAR_FLAGS_ABORT (1 << 5)
+// can be used alongside foreign to mark a foreign variable as writable
+#define SHVAR_FLAGS_FOREIGN_WRITABLE (1 << 6)
```
Comment (sinkingsugar) [inline include/shards/shards.h]: But foreign can be written by default, should be fine.

File: shards/modules/inputs/inputs.cpp [hunk 2]
```diff
@@ -620,4 +668,6 @@ SHARDS_REGISTER_FN(inputs) {
   using KeyUp = KeyUpDown<false>;
   REGISTER_SHARD("Inputs.KeyDown", KeyDown);
   REGISTER_SHARD("Inputs.KeyUp", KeyUp);
+
+  REGISTER_SHARD("Inputs.MatchModifierKeys", MatchModifierKeys);
```
Comment (sinkingsugar) [inline shards/modules/inputs/inputs.cpp]: A bit long, `MatchModifier` should be enough , we already have `Input` as context

File: shards/rust/src/core.rs [hunk 3]
```diff
@@ -485,7 +485,11 @@ impl WireRef {
       len: name.len() as u64,
     };
     unsafe {
-      (*Core).setExternalVariable.unwrap()(self.0, name, &var.0 as *const _ as *mut _);
+      let ev = SHExternalVariable {
+        var: &var.0 as *const _ as *mut _,
+        type_: std::ptr::null(),
```
Comment (sinkingsugar) [inline shards/rust/src/core.rs:490]: is that ok?

**Review by sinkingsugar — CHANGES_REQUESTED** [review]
LGTM just some minor requests

File: include/shards/shards.h (hunk 1 above)
Comment (guusw) [reply include/shards/shards.h]: No there's an assert in update I believe

File: shards/rust/src/core.rs (hunk 3 above)
Comment (guusw) [reply shards/rust/src/core.rs:490]: It's optional, null will still derive the type as before

File: shards/core/runtime.cpp [hunk 4]
```diff
@@ -1936,7 +1943,9 @@ NO_INLINE void _destroyVarSlow(SHVar &var) {
 }
 
 NO_INLINE void _cloneVarSlow(SHVar &dst, const SHVar &src) {
-  shassert((dst.flags & SHVAR_FLAGS_FOREIGN) != SHVAR_FLAGS_FOREIGN && "cannot clone into a foreign var");
```
Comment (guusw) [inline shards/core/runtime.cpp]: @sinkingsugar this one

Comment (sinkingsugar) [reply shards/core/runtime.cpp]: Ah true, but then it's external?

File: shards/modules/inputs/inputs.cpp (hunk 2 above)
Comment (guusw) [reply shards/modules/inputs/inputs.cpp]: Ok

File: shards/core/runtime.cpp (hunk 4 above)
Comment (guusw) [reply shards/core/runtime.cpp]: Changed it

**Comment by guusw** [comment]
Merged manually

**Pull request closed without merging**